### PR TITLE
COMP: Update platform check based on latest requirements

### PR DESCRIPTION
**Pull request by jamesobutler**
I've updated some platform checks to latest requirements. These checks were very very out of date.

### CODE REVIEW

File: CMake/UseSlicer.cmake.in [hunk 1]
```diff
@@ -208,7 +195,6 @@ if(NOT ${Slicer_SKIP_SET_CMAKE_C_CXX_FLAGS})
   set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS_INIT} ${Slicer_REQUIRED_C_FLAGS} ${ITK_REQUIRED_C_FLAGS} ${ADDITIONAL_C_FLAGS}" CACHE STRING "CMake C Flags" FORCE)
   set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS_INIT} ${Slicer_REQUIRED_CXX_FLAGS} ${ITK_REQUIRED_CXX_FLAGS} ${ADDITIONAL_CXX_FLAGS}" CACHE STRING "CMake CXX Flags" FORCE)
```
Comment (jcfr) [inline CMake/UseSlicer.cmake.in:196]: The modules was taking care of initializing `ITK_REQUIRED_C_FLAGS` and `ITK_REQUIRED_CXX_FLAGS`, we should check which flags were still being set on each platform and set them in https://github.com/Slicer/Slicer/blob/master/CMake/SlicerBlockCXXRequiredFlags.cmake

See also https://github.com/Slicer/Slicer/blob/117185ded4709017ec6b2c43d54bc0e1c2d053d9/CMakeLists.txt#L926-L927

**Review by jcfr — CHANGES_REQUESTED** [review]

File: CMake/UseSlicer.cmake.in (hunk 1 above)
Comment (jamesobutler) [reply CMake/UseSlicer.cmake.in]: I will pick off this `COMP: Remove outdated ITK platform specific checks` commit as it appeared to have a lot of outdated compiler options, but I'm currently unable to review these at the moment. Choosing to pick off to not hold up other commits in this PR.

File: CMake/SlicerBlockPlatformCheck.cmake [hunk 2]
```diff
@@ -34,42 +34,26 @@ endif()
...
-  #  12.x == Mac OSX 10.8 (Mountain Lion)
-  if(DARWIN_MAJOR_VERSION LESS "9")
-    message(FATAL_ERROR "Only Mac OSX >= 10.5 is supported !")
+  # and https://en.wikipedia.org/wiki/Darwin_(operating_system)#Release_history
+  #  17.x == macOS 10.13 (High Sierra)
+  #  18.x == macOS 10.14 (Mojave)
+  #  19.x == macOS 10.15 (Catalina)
+  #  20.x == macOS 11 (Big Sur)
+  #  21.x == macOS 12 (Monterey)
+  if(DARWIN_MAJOR_VERSION LESS "17")
```
Comment (jcfr) [inline CMake/SlicerBlockPlatformCheck.cmake:43]:
```suggestion
  if(NOT DARWIN_MAJOR_VERSION GREATER_EQUAL "17")
```

File: CMake/SlicerBlockPlatformCheck.cmake [hunk 3]
```diff
@@ -34,42 +34,26 @@ endif()
...
-    #   1500 = VS  9.0 (Visual Studio 2008)
-    #   1600 = VS 10.0 (Visual Studio 2010)
-    #   1700 = VS 11.0 (Visual Studio 2012)
-    if(MSVC_VERSION VERSION_LESS 1500)
-      message(FATAL_ERROR "Visual Studio >= 2008 is required !")
+    #   1900 = VS 14.0 (Visual Studio 2015)
+    #   1910-1919 = VS 15.0 (Visual Studio 2017)
+    #   1920-1929 = VS 16.0 (Visual Studio 2019)
+    #   1930-1939 = VS 17.0 (Visual Studio 2022)
+    if(MSVC_VERSION VERSION_LESS 1900)
```
Comment (jcfr) [inline CMake/SlicerBlockPlatformCheck.cmake:54]:
```suggestion
  if(NOT MSVC_VERSION VERSION_GREATER_EQUAL 1900)
```

File: CMake/SlicerBlockPlatformCheck.cmake (hunk 2 above)
Comment (jcfr) [reply CMake/SlicerBlockPlatformCheck.cmake:43]: Make use of new expressions supported by the `if` command. See https://cmake.org/cmake/help/v3.13/command/if.html

The test now directly match the message reported to the user.

File: CMake/SlicerBlockPlatformCheck.cmake (hunk 3 above)
Comment (jamesobutler) [reply CMake/SlicerBlockPlatformCheck.cmake]: I have update the syntax based on this recommendation

**Pull request merged**